validphys2/src/validphys/n3fit_data.py

```diff
@@ -92,7 +92,10 @@ def tr_masks(data, replica_trvlseed):
         cuts = dataset.cuts
         ndata = len(cuts.load()) if cuts else dataset.commondata.ndata
         frac = dataset.frac
-        trmax = int(frac * ndata)
+        if ndata == 1:
+            trmax = int(np.random.rand() < 0.75)
+        else:
+            trmax = int(frac * ndata)
```

Comment:
```suggestion
        trmax = int(frac*ndata)
        if ndata == 1:
            trmax = int(np.random.rand() <= frac)
```

I'll merge it after the bot is done. Let's see what happens to the two 1-point datasets.

Why would you want to do it for all datasets @RoyStegeman ?

Reply:
Why not? I know it probably won't make a difference in practice, but at least we'd be consistent in the treatment of all dataset, and technically it's what we should do anyway to really get the correct fraction.